### ISSUE TITLE
Support new OS - RH8 and Debian 10 (Buster) 

### ISSUE THREAD

**Issue by KChandrashekhar** [saltstack/salt#52590]
### Description of Epic 
Debian 10 (buster) is expected to release by mid-2019 (no exact release dates are available)
RH8 is also expected to release in May-June.

This epic groups all tasks that are necessary to get salt supported on above platforms

**Comment by clayoster** [comment]
Debian has announced that the release date for Buster is July 6th, 2019.

https://lists.debian.org/debian-devel-announce/2019/06/msg00003.html

**Comment by hbokh** [comment]
So is there an ETA for the "10" repo at `https://repo.saltstack.com/apt/debian/`?

**Comment by dmurphy18** [comment]
@hbokh There will be support in the next point release, Salt 2019.2.1 which is currently under going QA, however no date can be announced at this point.

**Comment by bryceml** [comment]
@hbokh We are waiting on debian 10 to available as an ami on AWS.  We are trying to avoid doing something custom in order to get it in quickly, and then have to go back and rework it when an official ami is relelased.  If it ends up taking too long waiting for an official ami, we may do something different.

**Comment by jandd** [comment]
The ami is prepared but not uploaded to the marketplace yet. https://lists.debian.org/debian-cloud/2019/07/msg00041.html

**Comment by frenkye** [comment]
Update on the repository?

**Comment by bryceml** [comment]
https://github.com/saltstack/os-imager/pull/120
https://github.com/saltstack/sre-jenkins/pull/180
https://github.com/saltstack/os-imager/pull/121

**Comment by kiemlicz** [comment]
Based on 
https://groups.google.com/forum/#!topic/salt-announce/5RJjtDYszx8

The 2019.2.1 release can take another months (I know that for Debian people like me, the eternal wait is the typical state of matters. However Debian 10 simplifies a lot of hacks required to properly install Salt which may have impact on day-to-day work) 

Thus is there any chance to backport Debian 10 support to 2019.2.0?

**Comment by boltronics** [comment]
I understand that fixing tests are important, but it's not a good look when there is no release compatible with the latest stable GNU/Linux distribution that runs such a [significant number](https://en.wikipedia.org/wiki/Usage_share_of_operating_systems#Market_share_by_category) of servers almost 2 months after release.

My workplace and one of my clients both have put plans to upgrade servers on hold, and I'm being asked when Salt is going to be ready.

**Comment by kiemlicz** [comment]
and so we wait
@Ch3LL is there any chance that you could provide some rough estimate of the 2019.2.1 release date (it doesn't have to be precise, even: weeks/months will do)?

**Comment by bensteinberg** [comment]
The buster AMI is released. From https://wiki.debian.org/Cloud/AmazonEC2Image/Buster, which includes AMI IDs:

> The initial publication of official Debian 10 (buster) AMIs for Amazon EC2 took place on 9 September, 2019. .... These images are not yet available on the AWS Marketplace, but that is planned.

**Comment by kiemlicz** [comment]
Could you please explain how does the AMI Debian 10 images relate to bootstrap script supporting Debian10?

**Comment by bensteinberg** [comment]
See the comment by @bryceml from July 22:

> @hbokh We are waiting on debian 10 to be available as an ami on AWS. We are trying to avoid doing something custom in order to get it in quickly, and then have to go back and rework it when an official ami is relelased. If it ends up taking too long waiting for an official ami, we may do something different.

**Comment by kiemlicz** [comment]
(I simply don't understand it)

**Comment by bryceml** [comment]
@kiemlicz The bootstrap script needs packages to be in the repo before it can install them, those are in staging and not released just quite yet.  We ended up building our own ami for testing with and will use the official one soon.  The ami's at this point are un-related to releasing, we just need to finish up preping the repo and updating the bootstrap script.

**Comment by boltronics** [comment]
Salt 2019.2.1 is out. Time to start testing with Buster. :tada:

**Comment by Reiner030** [comment]
@boltronics => not useable as shown in https://github.com/saltstack/salt/issues/54755

And there is no repository for buster available:

https://repo.saltstack.com/apt/debian/10/amd64/archive/
=> Not Found
=> The requested URL /apt/debian/10/amd64/archive/ was not found on this server.

**Comment by boltronics** [comment]
@Reiner030 Hmm... I'm not too surprised actually, since I saw the notification to package maintainers mail-list was sent out just 3 minutes prior to the public announcement post!

Considering this open issue, I have no idea why release announcements proceeded ahead of giving package maintainers time to do their thing. As it stands, the linked instructions on https://repo.saltstack.com/#debian won't work since (as you correctly pointed out) there is no https://repo.saltstack.com/apt/debian/10/ section yet.

Normally I build my own Debian packages and host my own private repository to minimize such issues with the environments I manage, but it would be nice if I didn't have to to start general testing on a Debian 10 box straight away.

**Comment by captnfab** [comment]
It seems this bugs comes from `salt/modules/lxc.py` in the _LXCConfig's method tempfile.
I tried to replace
        ntf = tempfile.NamedTemporaryFile()
        ntf.write(self.as_string())
        ntf.flush()
        return ntf
with:
        ntf = tempfile.NamedTemporaryFile()
        ntf.write(salt.utils.stringutils.to_bytes(self.as_string()))
        ntf.flush()
        return ntf

And it works… until the next error.

**Comment by captnfab** [comment]
Packages are available and installable now.

**Comment by bryceml** [comment]
https://github.com/saltstack/salt-jenkins/pull/1611 is blocked by the bootstrap script not supporting centos 8.

**Comment by ookull** [comment]
Support for Debian 10 packages is removed.

```
# apt update
...
E: The repository 'http://repo.saltstack.com/py3/ebian/10/amd64/latest buster Release' no longer has a Release file.
```

**Comment by ppmathis** [comment]
@ookull https://repo.saltstack.com/ explains the reason for doing so. To sum it up: 2019.2.1 is an unreliable release with several critical issues which is why it got removed from the repositories. An inelegant solution for Debian Buster though, as this was also the first version to be supported.

You could either wait for 2019.2.2 which is due to release soon™ or switch your repository URL temporarily to `https://repo.saltstack.com/py3/debian/10/amd64/archive/2019.2.1/`.

**Comment by bryceml** [comment]
last epic issue is #52837

**Comment by astronouth7303** [comment]
Trigger happy much, stale bot?

**Comment by sagetherage** [comment]
the last ticket in this epic is odd and will need to wait to be deprecated, so I am removing it from this epic so we can close it as done and will add it to the Sodium Epic for deprecation, but it may not be until Magnesium, we will determine that in planning in early Feb 2020.
